code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/ServiceActivate.java

```diff
@@ -68,15 +67,40 @@
     @Inject
     ServiceDiscoveryService sdServer;
 
+    @Inject
+    ServiceExposeMapDao svcExposeDao;
+
     @Override
     public String[] getProcessNames() {
-        return new String[] { ServiceDiscoveryConstants.PROCESS_SERVICE_ACTIVATE };
+        return new String[] { ServiceDiscoveryConstants.PROCESS_SERVICE_ACTIVATE,
+                ServiceDiscoveryConstants.PROCESS_SERVICE_UPDATE };
     }
 
     @Override
     @SuppressWarnings("unchecked")
     public HandlerResult handle(ProcessState state, ProcessInstance process) {
         Service service = (Service) state.getResource();
+
+        // on inactive service update, do nothing
+        if (process.getName().equalsIgnoreCase(ServiceDiscoveryConstants.PROCESS_SERVICE_UPDATE)
```

Comment:
@cjellick well, I just couldn't find a reason of why disallow it :) its just a simple object's property change when inactive. On the service activation, number of service's instances will be = scale